bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -530,27 +550,28 @@ void getNextData(SerializedData current)
     public long rewind(long windowId)
     {
       long bs = starting_window & 0x7fffffff00000000L;
-      DataListIterator dli = getIterator(this);
-      done:
-      while (dli.hasNext()) {
-        SerializedData sd = dli.next();
-        switch (sd.buffer[sd.dataOffset]) {
-          case MessageType.RESET_WINDOW_VALUE:
-            ResetWindowTuple rwt = (ResetWindowTuple)Tuple.getTuple(sd.buffer, sd.dataOffset, sd.length - sd.dataOffset + sd.offset);
-            bs = (long)rwt.getBaseSeconds() << 32;
-            if (bs > windowId) {
-              writingOffset = sd.offset;
-              break done;
-            }
-            break;
+      try (DataListIterator dli = getIterator(this)) {
+        done:
+        while (dli.hasNext()) {
+          SerializedData sd = dli.next();
+          switch (sd.buffer[sd.dataOffset]) {
+            case MessageType.RESET_WINDOW_VALUE:
+              ResetWindowTuple rwt = (ResetWindowTuple) Tuple.getTuple(sd.buffer, sd.dataOffset, sd.length - sd.dataOffset + sd.offset);
```

Comment:
Please clarify what's wrong with the existing format. Note that the only real change here is try-with-resources.